pkg/cri/resource-manager/control/blockio/blockio.go

```diff
@@ -133,25 +164,282 @@ func (ctl *blockio) BlockIOClass(c cache.Container) string {
+							LeafWeight: leafWeight})
+				}
+				if throttleReadBps != -1 {
+					oci.ThrottleReadBpsDevice = append(oci.ThrottleReadBpsDevice,
+						cgroups.OciRateDeviceParameters{
+							Major: blockDeviceInfo.Major,
+							Minor: blockDeviceInfo.Minor,
+							Rate:  throttleReadBps})
+				}
+				if throttleWriteBps != -1 {
+					oci.ThrottleWriteBpsDevice = append(oci.ThrottleWriteBpsDevice,
+						cgroups.OciRateDeviceParameters{
+							Major: blockDeviceInfo.Major,
+							Minor: blockDeviceInfo.Minor,
+							Rate:  throttleWriteBps})
+				}
+				if throttleReadIOPS != -1 {
+					oci.ThrottleReadIOPSDevice = append(oci.ThrottleReadIOPSDevice,
+						cgroups.OciRateDeviceParameters{
+							Major: blockDeviceInfo.Major,
+							Minor: blockDeviceInfo.Minor,
+							Rate:  throttleReadIOPS})
+				}
+				if throttleWriteIOPS != -1 {
+					oci.ThrottleWriteIOPSDevice = append(oci.ThrottleWriteIOPSDevice,
+						cgroups.OciRateDeviceParameters{
+							Major: blockDeviceInfo.Major,
+							Minor: blockDeviceInfo.Minor,
+							Rate:  throttleWriteIOPS})
+				}
+			}
+		}
+		err = utils.JoinErrors(errors)
+		if err != nil {
+			return oci, err
+		}
+	}
+	return oci, nil
+}
+
+func parseAndValidateInt64(fieldName string, fieldContent string,
+	defaultValue int64, min int64, max int64) (int64, error) {
+	// Returns field content
+	if fieldContent == "" {
+		return defaultValue, nil
+	}
+	qty, err := resource.ParseQuantity(fieldContent)
+	if err != nil {
+		return defaultValue, fmt.Errorf("syntax error in %#v (%#v)", fieldName, fieldContent)
+	}
+	value := qty.Value()
+	if min != -1 && min > value {
+		return defaultValue, fmt.Errorf("value of %#v (%#v) smaller than minimum (%#v)", fieldName, value, min)
+	}
+	if max != -1 && value > max {
+		return defaultValue, fmt.Errorf("value of %#v (%#v) bigger than maximum (%#v)", fieldName, value, max)
+	}
+	return value, nil
+}
+
+type BlockDeviceInfo struct {
+	Major  int64
+	Minor  int64
+	Origin string // for traceability: why this device is configured
+}
+
+func configurableBlockDevices(devWildcards []string) (map[string]BlockDeviceInfo, error) {
+	// Return map {devNode: BlockDeviceInfo}
+	// Example: {"/dev/sda": {Major:8, Minor:0, Origin:"from symlink /dev/disk/by-id/ata-VendorXSSD from wildcard /dev/disk/by-id/*SSD*"}}
+	errors := []error{}
+	blockDevices := map[string]BlockDeviceInfo{}
+	var origin string
+
+	// 1. Expand wildcards to device filenames (may be symlinks)
+	// Example: devMatches["/dev/disk/by-id/ata-VendorSSD"] == "from wildcard \"dev/disk/by-id/*SSD*\""
+	devMatches := map[string]string{} // {devNodeOrSymlink: origin}
+	for _, devWildcard := range devWildcards {
+		devWildcardMatches, err := filepath.Glob(devWildcard)
+		if err != nil {
+			errors = append(errors, fmt.Errorf("bad device wildcard %#v: %+v", devWildcard, err))
+			continue
+		}
+		if len(devWildcardMatches) == 0 {
+			log.Warn("deviceParametersToOci: device wildcard %#v does not match any device nodes", devWildcard)
```

Comment:
nitpick: isn't the 'function name prefix:' in the warning message incorrect (shouldn't it be `configurableBlockDevices`) ?

Reply:
Yep that was wrong. Removed the prefix completely for better error messages.